src/ExtractTileOperations.cpp

```diff
@@ -0,0 +1,414 @@
+#include "ExtractTileOperations.h"
+
+#include "IRMatch.h"  // expr_match
+#include "IRMutator.h"
+#include "IROperator.h"  // Expr + Expr
+#include "Util.h"        // ScopedValue
+
+namespace Halide {
+namespace Internal {
+
+using std::string;
+using std::vector;
+
+namespace {
+
+template<int Dim>
+struct Tile {
+    bool result;
+    Expr base;
+    Expr stride[Dim];
+    int extent[Dim];
+};
+
+enum class AMXOpType {
+    Int8,
+    Bf16,
+};
+
+/// returns the appropriate `Halide::Type` for the given operation type
+Type amx_op_type_result_type(AMXOpType op_ty) {
+    switch (op_ty) {
+    case AMXOpType::Int8:
+        return Int(32, 256);
+    case AMXOpType::Bf16:
+        return Float(32, 256);
+    default:
+        return Type();
+    }
+}
+
+const auto wild_i32 = Variable::make(Int(32), "*");
+const auto wild_i32x = Variable::make(Int(32, 0), "*");
+
+Tile<2> is_2d_tile_index(const Expr &e) {
```

Comment:
Nit: I'd expect a function named "is_whatever" to return bool, but this returns a struct. Something like `get_2d_tile_index` would be better.